build_tools/kokoro/gcp_ubuntu/cmake/linux/x86-swiftshader-asan/build.sh

```diff
@@ -52,8 +52,10 @@ echo "------------------"
 # Respect the user setting, but default to as many jobs as we have cores.
 export CTEST_PARALLEL_LEVEL=${CTEST_PARALLEL_LEVEL:-$(nproc)}
 
-# Respect the user setting, but default to turning on vulkan and llvmaot.
-export IREE_VULKAN_DISABLE=${IREE_VULKAN_DISABLE:-0}
+# Respect the user setting, but default to turning off the vulkan tests
+# and turning on the llvmaot ones.
+# TODO(#5716): Fix and enable Vulkan lit tests.
+export IREE_VULKAN_DISABLE=${IREE_VULKAN_DISABLE:-1}
```

Comment:
Ah... this also sets `label_exclude_args+=("^driver=vulkan$")`, which disables the HAL tests too. We could split that out. WDYT?

Reply:
Seems to pass presubmit, at least: https://source.cloud.google.com/results/invocations/dc784279-e4ed-43e4-ac90-a78fd83c9113/targets/iree%2Fgcp_ubuntu%2Fcmake%2Flinux%2Fx86-swiftshader-asan%2Fpresubmit/log

```
iree/hal/vulkan                                                  =   0.16 sec*proc (1 test)
iree/hal/vulkan/cts                                              =  13.87 sec*proc (11 tests)
```

(compared to without the split: [logs](https://source.cloud.google.com/results/invocations/01e100ef-a8a7-4a55-80b5-73cd26a204c4/targets/iree%2Fgcp_ubuntu%2Fcmake%2Flinux%2Fx86-swiftshader-asan%2Fpresubmit/log))

Reply:
That was deliberate. Early on, I tried to make our CI scripts runnable locally with minimal friction. The idea was that someone could have `IREE_VULKAN_DISABLE` in their env and turn off all the Vulkan tests. I'm not sure that use case ever really worked well though. If you *do* think we should change the way these env variables work with build scripts, please change all the build scripts to stay consistent.

Reply:
How's this? Also...

* Is `IREE_LLVMAOT_DISABLE` still useful?
* Should I split `IREE_CUDA_DISABLE` as well?

(at some point I just want to get the CI green again and changes can be made in follow-ups - maybe we should just revert https://github.com/google/iree/commit/e35d00338a3ac9f8943a5f7050c632939ebe14cb?)

Reply:
I can't remember why `IREE_LLVMAOT_DISABLE` was ever necessary, since I'm not sure how we could have the LLVM_AOT runtime not available. Probably from a time when the driver was experimental and buggy or something.

> (at some point I just want to get the CI green again and changes can be made in follow-ups - maybe we should just revert e35d003?)

Yeah I think this is getting too complicated. I would just revert. Actually better to set `IREE_VULKAN_DISABLE` (it was previously accidentally hardcoded) as you originally had in this PR.

Reply:
Looks like you added it in https://github.com/google/iree/commit/aaa08b1ec83b201e765950ca417ebd22519d4e08 and it replaced `IREE_LLVMJIT_DISABLE` which is much more understandable as a thing that would only work on some platforms. I think deleting it would probably be fine, but let's do that separately.

Reply:
> Yeah I think this is getting too complicated. I would just revert. Actually better to set IREE_VULKAN_DISABLE (it was previously accidentally hardcoded) as you originally had in this PR.

Ok, sent https://github.com/google/iree/pull/8550 as a simpler revert + small fix.

I can take a more detailed look at these scripts after.